### PR TITLE
Update Sources to http://download.qt.io/snapshots/ci/qtwebkit/

### LINKED ISSUE #11: build linux package
Is it somehow possible to enable linux for this feedstock and update to qt5.12?

### PR DESCRIPTION
**Pull request by gillins**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

This PR attempts to get `qtwebkit` building again by using the files provided under http://download.qt.io/snapshots/ci/qtwebkit/ as suggested by @ThomasG77 (https://github.com/conda-forge/qgis-feedstock/issues/68#issuecomment-720162003). Seems this project is still alive, and needed by QGIS.

Fixes #11 #12 and #13. 

xref: https://github.com/conda-forge/qgis-feedstock/issues/68

### CODE REVIEW

**Comment by gillins** [comment]
@conda-forge-admin, please rerender

**Comment by gillins** [comment]
@conda-forge-admin, please rerender

**Comment by gillins** [comment]
@conda-forge-admin, please rerender

**Comment by gillins** [comment]
@conda-forge-admin, please rerender

**Comment by gillins** [comment]
@conda-forge-admin, please rerender

**Comment by gillins** [comment]
@conda-forge-admin, please rerender

**Comment by gillins** [comment]
@conda-forge-admin, please rerender

**Comment by gillins** [comment]
@conda-forge-admin, please rerender

**Comment by gillins** [comment]
@conda-forge-admin, please rerender

**Comment by gillins** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,37 +1,96 @@
...
+    - m2-bison  # [win]
+    - bison  # [unix]
+    - m2-flex  # [win]
+    - flex  # [unix]
+    - m2-gperf  # [win]
+    - gperf  # [unix]
   host:
-    # qtwebkit needs to be used against the same version of Qt that
-    # it was released for.
-    - qt {{ ver }}
```
Comment (gillins) [inline recipe/meta.yaml:23]: I used to think that `qtwebkit`'s version was tied to a particular `qt` version, but their release notes mention a range of `qt` versions so I now think it is independent. Does anyone know more?

File: recipe/0002-win-gperf-lf.patch [hunk 2]
```diff
@@ -0,0 +1,63 @@
+--- Source/WebCore/platform/network/create-http-header-name-table.orig	2020-11-24 14:11:54.102671419 +1000
```
Comment (gillins) [inline recipe/0002-win-gperf-lf.patch:1]: Is is possible to make `m2-gperf` understand Windows line endings? Or do we need a MSVC version of `gperf`? Would be nice to remove these patches.

File: recipe/meta.yaml (hunk 1 above)
Comment (ocefpaf) [reply recipe/meta.yaml:23]: Me too. If that is not true let's unpin this.

**Comment by gillins** [comment]
@SrNetoChan Are you happy if I merge this. Not really a 'final' release, but seems to work ok...

**Comment by SrNetoChan** [comment]
Fine by me. Then we can try using it for QGIS.

**Comment by SrNetoChan** [comment]
Thanks for this work. It would be great to see this working on QGIS, which is currently the major limitation.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,37 +1,96 @@
...
+    - {{ cdt('libxcb') }}                # [linux]
+    - {{ cdt('libxext') }}               # [linux]
+    - {{ cdt('libxau') }}                # [linux]
+    - {{ cdt('libxcomposite-devel') }}   # [linux]
+    - cmake
+    - ninja
+    # On Linux, cmake picks up /usr/bin/python2 (since python2 is checked
+    # first) which is v2.6 and too old. Workaround this by giving it a
+    # new Python 2
+    - python 2.7  # [linux]
```
Comment (SrNetoChan) [inline recipe/meta.yaml:46]: Will this conflict with QGIS Python 3.x?

**Review by ocefpaf — APPROVED** [review]

**Review by SrNetoChan — APPROVED** [review]

File: recipe/meta.yaml (hunk 3 above)
Comment (gillins) [reply recipe/meta.yaml:46]: It's just needed during the build of this package, so shouldn't interfere with Python 3.x at runtime.

File: recipe/0002-win-gperf-lf.patch (hunk 2 above)
Comment (gillins) [reply recipe/0002-win-gperf-lf.patch:1]: I actually looked into this and appears that `gperf` always assumes Unix line endings (see `getline.cc`). Not sure much can be done without major surgery on the sources. Will keep the patches now now...

**Pull request merged**